### DISCUSSION TITLE
How to support SharedArrayBuffer / enable crossOriginIsolated in Simple Browser

### DISCUSSION THREAD

**Discussion by devarend** [microsoft/vscode-discussions#978] [Extension Development QnA] [answered]
I'm building an extension which opens a Simple Browser using: 
`commands.executeCommand('simpleBrowser.show', Uri.parse(message.text));`

Which then runs an application that is using **SharedArrayBuffer**. It is working fine when I use an iframe (without using Simple Browser) but as soon as I try to open the running application in a separate tab using Simple Browser I run into the following error:

`Failed to execute 'postMessage' on 'Worker': SharedArrayBuffer transfer requires self.crossOriginIsolated.` 

Is it possible to use **SharedArrayBuffer** in a Simple Browser tab or do I have to stick to my iframe?

**Comment by tamuratak** [comment, marked as answer]
> Is it possible to use SharedArrayBuffer in a Simple Browser tab 

I think it is not possible. Simple Browser is implemented with iframe. And, the content of the iframe interacts with the parent of the iframe for the features of Simple Browser. The iframe is not isolated, and `self.crossOriginIsolated` must be false.

- https://developer.mozilla.org/en-US/docs/Web/API/crossOriginIsolated
- https://github.com/microsoft/vscode/blob/30b777312745e84972956d4361465d4d38aa0f78/extensions/simple-browser/src/simpleBrowserView.ts#L168

**Reply by devarend** [reply]
Thanks, fixed it by using my own webview with an iframe that does allow `crossOriginIsolated`.
